src/common/tiglcommonfunctions.cpp

```diff
@@ -1327,6 +1327,122 @@ TopoDS_Wire BuildWireRectangle(const double heightToWidthRatio, const double cor
     return wire;
 }
 
+namespace
+{
+    class SuperEllipse : public tigl::MathFunc3d
+    {
+    public:
+        SuperEllipse(const double lowerHeightFraction, const double mLower, const double mUpper,
+                const double nLower, const double nUpper)
+            : tigl::MathFunc3d(), m_lowerHeightFraction(lowerHeightFraction), m_mLower(mLower),
+            m_mUpper(mUpper), m_nLower(nLower), m_nUpper(nUpper) {}
+
+        double valueX(double t) override
+        {
+            return 0.;
+        }
+
+        double valueY(double t) override
+        {
+            //curve is built clockwise
+            return 0.5*std::sin(t);
+        }
+
+        double valueZ(double t) override
+        {
+            double z_0 = m_lowerHeightFraction - 0.5;
+            //clean angle from traversion factor (ensures that alpha <= 2*PI, to determine quadrant)
+            double alpha = (std::fmod(t,(2.*M_PI)));
+            double param = 0.5*std::sin(alpha);
+
+            //same order as in building the profile: oriented clockwise, starting with 1. quadrant ending with 2.
+
+            //build right upper half of semi ellipse
+            if((alpha>=0.)&&(alpha<M_PI/2.)){
+                double z = z_0 + (0.5 -z_0)* std::pow((1. - std::pow(std::abs(2. * param), m_mUpper)), 1. / m_nUpper);
+                return z;
+            }
+
+            //build lower right half of semi ellipse
+            if((alpha>=M_PI/2.)&&(alpha<M_PI)){
+                double z = z_0 - (0.5 + z_0) * std::pow((1. - std::pow(std::abs(2. * param), m_mLower)), 1. / m_nLower);
+                return z;
+            }
+
+            //build lower left half of semi ellipse
+            if((alpha>=M_PI)&&(alpha<M_PI*3./2.)){
+                double z = z_0 - (0.5 + z_0) * std::pow((1. - std::pow(std::abs(2. * param), m_mLower)), 1. / m_nLower);
+                return z;
+            }
+
+            //build left upper half of semi ellipse
+            if((alpha>=M_PI*3./2.)&&(alpha<=M_PI*2.)){
+                double z = z_0 + (0.5 -z_0)* std::pow((1. - std::pow(std::abs(2. * param), m_mUpper)), 1. / m_nUpper);
+                return z;
+            } else {
+                throw tigl::CTiglError("Error building Ellipse");
+            }
+
+        }
+
+    private:
+          double m_lowerHeightFraction;
+          double m_mLower;
+          double m_mUpper;
+          double m_nLower;
+          double m_nUpper;
+    };
+
+} //anonymos namespace
+
+TIGL_EXPORT TopoDS_Wire BuildWireSuperEllipse(const double lowerHeightFraction, const double mLower, const double mUpper,
+                                              const double nLower, const double nUpper, const double tol)
+{
+    if (mLower==0||mUpper==0||nLower==0||nUpper==0||lowerHeightFraction==0||lowerHeightFraction==1){
+        throw tigl::CTiglError("Invalid input. Check superellipse profile parameters.");
+    }
```

Comment:
I would prefer it if we were more verbose about the error. We could split this into exponents and then throw a message stating that the exponents may not be 0 and into lowerHeightFraction. 

In addition, we are only checking edge cases of floating point numbers with equality here. We should check with tolerances or probably better, with inequalities: What happens if an exponent is negative? What happens if lowerHeightFraction is larger than 1 or smaller than zero? Wouldn't it make sense to throw an exception for 0 or negative exponents and an exception for lowerHeightFractions smaller or equal to zero and larger or equal to one?